src/stores/AccountController.ts

```diff
@@ -104,7 +104,11 @@ export async function updateUser(values: { firstName: string; lastName: string }
   await withAccountService(async ({ accountService, sandbox }) => {
     useAccountStore.setState({ loading: true });
 
-    const { auth, user } = useAccountStore.getState();
+    const { auth, user, canUpdateEmail } = useAccountStore.getState();
+
+    if (Object.prototype.hasOwnProperty.call(values, 'email') && !canUpdateEmail) {
+      throw new Error('Email update not supported');
```

Comment:
I don't think that we need this validation here

Reply:
Never hurts to have sanity checks. Maybe the code is written so that this can never happen now, but what about when someone refactors later?

Reply:
ok, then, we can go with this check :)